source/puppet/4.4/reference/system_requirements.markdown

```diff
@@ -4,87 +4,78 @@ title: "Puppet 4.4 System Requirements"
 canonical: "/puppet/latest/reference/system_requirements.html"
 ---
 
-> To install Puppet, first [view the pre-install tasks](./install_pre.html).
+[pe-requirements]: {{pe}}/install_system_requirements.html#supported-operating-systems
+[preinstall-network]: ./install_pre.html#check-your-network-configuration
 
-## Hardware
+Puppet can run on many operating systems and hardware configurations. Before installing Puppet, first [view the pre-install tasks](./install_pre.html).
 
-The Puppet agent service has no particular hardware requirements and can run on nearly anything.
+> **Note:** For [Puppet Enterprise's](/pe/) supported platforms, see its [system requirements][pe-requirements].
 
-However, the Puppet master service is fairly resource intensive, and should be installed on a robust dedicated server.
+## Puppet agent
 
-* At a minimum, your Puppet master server should have two processor cores and at least 1 GB of RAM.
-* To comfortably serve at least 1,000 nodes, it should have 2-4 processor cores and at least 4 GB of RAM.
+The Puppet agent service is lightweight enough to run on nearly any system capable of running a Ruby interpreter. To be managed by a Puppet master, the agent must be accessible from the master. See the [pre-installation instructions][preinstall-network] for details about the required DNS and port configuration for agents.
```

Comment:
@kylog Do we have official system requirements for Puppet agent?

Also, should we continue to note requirements for the Ruby master, and if so, can I confirm what they are?